state/executor.go

```diff
@@ -510,10 +531,25 @@ func (t *Transition) apply(msg *types.Transaction) (*runtime.ExecutionResult, er
 	remaining := new(big.Int).Mul(new(big.Int).SetUint64(result.GasLeft), msg.GasPrice)
 	t.state.AddBalance(msg.From, remaining)
 
-	// pay the coinbase
-	coinbaseFee := new(big.Int).Mul(new(big.Int).SetUint64(result.GasUsed), msg.GasPrice)
+	// define effective tip based on tx type
+	effectiveTip := msg.GasPrice
```

Comment:
maybe it would be good to explain calculations of miner fee and burn amount with comment and link(s)?

Reply:
The calculation is not complex at all: we just use the gas fee cap and gas tip cap depending on the london hard fork enablement. Added more comments.